vizro-core/src/vizro/models/_components/button.py

```diff
@@ -84,16 +97,40 @@ def _action_outputs(self) -> dict[str, _IdProperty]:
     @_log_call
     def build(self):
         variants = {"plain": "link", "filled": "primary", "outlined": "secondary"}
-        description = self.description.build().children if self.description else [None]
+        description = self._build_description()
+        icon = (
+            html.Span(self.icon, id=f"{self.id}-icon", className="material-symbols-outlined tooltip-icon")
+            if self.icon
+            else None,
+        )
 
         defaults = {
             "id": self.id,
-            "children": html.Span([self.text, *description], className="button-text"),
+            "children": html.Span([*icon, self.text, *description], className="btn-text"),
             "href": get_relative_path(self.href) if self.href.startswith("/") else self.href,
             "target": "_top",
             # dbc.Button includes `btn btn-primary` as a class by default and appends any class names provided.
             # To prevent unnecessary class chaining, the button's style variant should be specified using `color`.
             "color": variants[self.variant],
+            "class_name": "btn-circular" if self.icon and not self.text else "",
         }
 
         return dbc.Button(**(defaults | self.extra))
+
+    def _build_description(self) -> list[Optional[Union[dbc.Tooltip, html.Span]]]:
+        """Conditionally returns the tooltip based on the provided `text` and `icon` arguments.
+
+        If text='', the tooltip icon is omitted, and the tooltip text is shown when hovering over the button icon.
+        Otherwise, the tooltip icon is displayed.
+        """
+        if not self.description:
+            return [None]
+
+        description = self.description.build().children
+        if not self.text:
+            # retrieves the dbc.Tooltip instance from the vm.Tooltip model to update its tooltip target
+            tooltip_component = description[1]
```

Comment:
```suggestion
            tooltip_component = description[description.id]
```

I think should work the same and would be more robust than referencing by position in a list.

This is generally nice code though, I like it 👍

Reply:
Oh sorry, I must be blind. I see now.